server/blob/blob-export-api/src/main/java/org/apache/james/blob/export/api/BlobExportMechanism.java

```diff
@@ -29,14 +29,33 @@ public BlobExportException(String message, Throwable cause) {
         }
     }
 
+    enum BlobFileExtension {
+        ZIP("zip");
+
+        private final String extension;
+
+        BlobFileExtension(String extension) {
+            this.extension = extension;
+        }
+
+        public String getExtension() {
+            return extension;
+        }
+    }
+
     @FunctionalInterface
     interface ShareeStage {
         ExplanationStage with(MailAddress sharee);
     }
 
     @FunctionalInterface
     interface ExplanationStage {
-        FinalStage explanation(String explanation);
+        BlobFileExtensionStage explanation(String explanation);
+    }
+
+    @FunctionalInterface
+    interface BlobFileExtensionStage {
+        FinalStage fileExtension(BlobFileExtension extension);
```

Comment:
IMO this stage should be optional - as the caller might not know the type of the data - in that case we should not have some extensions rather than a potentially wrong one.